### PR TITLE
feat(core): Support ResourceURI input for lock/unlock requests

### PR DESCRIPTION
**Pull request by apeteri**

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/request/ResourceLockRequestTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+					.execute(Services.bus())
+					.getSync();
+			});
+			
+			assertThat(lockedException.getMessage())
+				.isEqualTo("Could not acquire requested lock(s) while committing changes because %s is %s", user, lockDescription);
+			
+		} finally {
+			
+			LockRequests.prepareResourceLock()
```
Comment (nagyo) [inline snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/request/ResourceLockRequestTest.java]: Shouldn't this be a `prepareResourceUnlock()` ?

Comment (nagyo) [reply snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/request/ResourceLockRequestTest.java]: Would also be nice to test if the unlock worked or not

**Review by nagyo — APPROVED** [review]
LGTM, with a small note

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/request/ResourceLockRequestTest.java (hunk 1 above)
Comment (apeteri) [reply snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/request/ResourceLockRequestTest.java]: > Shouldn't this be a `prepareResourceUnlock()` ?

Indeed! 😊 I'll try to add the description again after unlocking the resource's working branch.

**Comment by nagyo** [comment]
Merging PR with an expected follow-up review from @cmark

**Review by cmark — COMMENTED** [review]
It looks great, but there is still work to do. The low-level repository + branchPath-based locking mechanism can be marked deprecated and removed from 9.0.

**Pull request merged**